### PR TITLE
[Mellanox] Add a new Mellanox platform x86_64-mlnx_msn3420 and new SKU ACS-MSN3420

### PR DESCRIPTION
**Pull request by shlomibitton**
- What I did
Add a new Mellanox platform x86_64-mlnx_msn3420 and new SKU ACS-MSN3420.

- How I did it

1.  Add related files to the device folder:

buffer config templates
pg lookup profile
port_config.ini
sai profile
sensor conf
plugins

2.  Edited related platform api files:

chassis.py
psu.py
thermal.py

- How to verify it

- Description for the changelog

- A picture of a cute animal (not mandatory but encouraged)

### CODE REVIEW

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py [hunk 1]
```diff
@@ -41,8 +41,8 @@
 
 # magic code defnition for port number, qsfp port position of each hwsku
 # port_position_tuple = (PORT_START, QSFP_PORT_START, PORT_END, PORT_IN_BLOCK, EEPROM_OFFSET)
-hwsku_dict = {'ACS-MSN2700': 0, 'Mellanox-SN2700': 0, 'Mellanox-SN2700-D48C8': 0, 'LS-SN2700': 0, 'ACS-MSN2740': 0, 'ACS-MSN2100': 1, 'ACS-MSN2410': 2, 'ACS-MSN2010': 3, 'ACS-MSN3700': 0, 'ACS-MSN3700C': 0, 'ACS-MSN3800': 4, 'Mellanox-SN3800-D112C8': 4, 'ACS-MSN4700': 0}
-port_position_tuple_list = [(0, 0, 31, 32, 1), (0, 0, 15, 16, 1), (0, 48, 55, 56, 1), (0, 18, 21, 22, 1), (0, 0, 63, 64, 1)]
+hwsku_dict = {'ACS-MSN2700': 0, 'Mellanox-SN2700': 0, 'Mellanox-SN2700-D48C8': 0, "LS-SN2700":0, 'ACS-MSN2740': 0, 'ACS-MSN2100': 1, 'ACS-MSN2410': 2, 'ACS-MSN2010': 3, 'ACS-MSN3420': 5, 'ACS-MSN3700': 0, 'ACS-MSN3700C': 0, 'ACS-MSN3800': 4, 'Mellanox-SN3800-D112C8': 4, 'ACS-MSN4700': 0}
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py:44]: why we have " " for LS-SN2700 and for all the rest ' '

Comment (liat-grozovik) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py:44]: why adding new system in the middle and not at the last of the dic? what is the value 5 added for SN3240 stands for?

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/buffers.json.j2 [hunk 2]
```diff
@@ -0,0 +1 @@
+../../x86_64-mlnx_msn2700-r0/ACS-MSN2700/buffers.json.j2
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/buffers.json.j2:1]: this is a Spectrum-2 switch, why buffers are taken as Spectrum switch? it should be more like SN3700

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/pg_profile_lookup.ini [hunk 3]
```diff
@@ -0,0 +1 @@
+../../x86_64-mlnx_msn2700-r0/ACS-MSN2700/pg_profile_lookup.ini
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/pg_profile_lookup.ini:1]: same here. use SN3700 as your base

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/port_config.ini [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+Ethernet200     200,201,202,203     etp51     50
+Ethernet204     204,205,206,207     etp52     51
+Ethernet208     208,209,210,211     etp53     52
+Ethernet212     212,213,214,215     etp54     53
+Ethernet216     216,217,218,219     etp55     54
+Ethernet220     220,221,222,223     etp56     55
+Ethernet224     224,225,226,227     etp57     56
+Ethernet228     228,229,230,231     etp58     57
+Ethernet232     232,233,234,235     etp59     58
+Ethernet236     236,237,238,239     etp60     59
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/port_config.ini:61]: should you have new line at the end of the file or not? not sure about it just double checking

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/qos.json.j2 [hunk 5]
```diff
@@ -0,0 +1 @@
+../../x86_64-mlnx_msn2700-r0/ACS-MSN2700/qos.json.j2
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/qos.json.j2:1]: same here, use SN3700 as base

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py (hunk 1 above)
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py:44]: we are using the same file for all platforms, thus adding a symlink to this file and adding the SKU to the dic.
value 5 stands for the "port_position_tuple_list" index.
its a mapping between the SKU and the "PORT_START, QSFP_PORT_START, PORT_END, PORT_IN_BLOCK, EEPROM_OFFSET"

regarding the new system in the middle, moved to the end.

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/buffers.json.j2 (hunk 2 above)
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/buffers.json.j2:1]: fixed

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/qos.json.j2 (hunk 5 above)
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/qos.json.j2:1]: fixed

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/port_config.ini (hunk 4 above)
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/port_config.ini:61]: fixed

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/pg_profile_lookup.ini (hunk 3 above)
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/pg_profile_lookup.ini:1]: fixed

**Comment by keboliu** [comment]
retest vsimage please

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 6]
```diff
@@ -49,8 +49,8 @@
 
 # magic code defnition for port number, qsfp port position of each hwsku
 # port_position_tuple = (PORT_START, QSFP_PORT_START, PORT_END, PORT_IN_BLOCK, EEPROM_OFFSET)
-hwsku_dict_port = {'ACS-MSN2010': 3, 'ACS-MSN2100': 1, 'ACS-MSN2410': 2, 'ACS-MSN2700': 0, 'Mellanox-SN2700': 0, 'Mellanox-SN2700-D48C8': 0, 'LS-SN2700':0, 'ACS-MSN2740': 0, 'ACS-MSN3700': 0, 'ACS-MSN3700C': 0, 'ACS-MSN3800': 4, 'Mellanox-SN3800-D112C8': 4, 'ACS-MSN4700': 0}
-port_position_tuple_list = [(0, 0, 31, 32, 1), (0, 0, 15, 16, 1), (0, 48, 55, 56, 1), (0, 18, 21, 22, 1), (0, 0, 63, 64, 1)]
+hwsku_dict_port = {'ACS-MSN2010': 3, 'ACS-MSN2100': 1, 'ACS-MSN2410': 2, 'ACS-MSN2700': 0, 'Mellanox-SN2700': 0, 'Mellanox-SN2700-D48C8': 0, 'LS-SN2700':0, 'ACS-MSN2740': 0, 'ACS-MSN3420': 5, 'ACS-MSN3700': 0, 'ACS-MSN3700C': 0, 'ACS-MSN3800': 4, 'Mellanox-SN3800-D112C8': 4, 'ACS-MSN4700': 0}
```
Comment (liat-grozovik) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:52]: why adding new system in the middle of the list and not at the end?

File: platform/mellanox/mlnx-platform-api/sonic_platform/psu.py [hunk 7]
```diff
@@ -37,15 +37,15 @@
 
 # in most SKUs the file psuX_curr, psuX_volt and psuX_power contain current, voltage and power data respectively. 
 # but there are exceptions which will be handled by the following dictionary
-hwsku_dict_psu = {'ACS-MSN3700': 1, 'ACS-MSN3700C': 1, 'ACS-MSN3800': 1, 'Mellanox-SN3800-D112C8': 1, 'ACS-MSN4700': 1}
+hwsku_dict_psu = {'ACS-MSN3420': 1, 'ACS-MSN3700': 1, 'ACS-MSN3700C': 1, 'ACS-MSN3800': 1, 'Mellanox-SN3800-D112C8': 1, 'ACS-MSN4700': 1}
```
Comment (liat-grozovik) [inline platform/mellanox/mlnx-platform-api/sonic_platform/psu.py:40]: same here. please add new systems at the end

Comment (shlomibitton) [reply platform/mellanox/mlnx-platform-api/sonic_platform/psu.py]: fixed

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py (hunk 6 above)
Comment (shlomibitton) [reply platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py]: fixed

**Comment by jleveque** [comment]
Retest vsimage please

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py [hunk 8]
```diff
@@ -41,8 +41,8 @@
 
 # magic code defnition for port number, qsfp port position of each hwsku
 # port_position_tuple = (PORT_START, QSFP_PORT_START, PORT_END, PORT_IN_BLOCK, EEPROM_OFFSET)
-hwsku_dict = {'ACS-MSN2700': 0, 'Mellanox-SN2700': 0, 'Mellanox-SN2700-D48C8': 0, 'LS-SN2700': 0, 'ACS-MSN2740': 0, 'ACS-MSN2100': 1, 'ACS-MSN2410': 2, 'ACS-MSN2010': 3, 'ACS-MSN3700': 0, 'ACS-MSN3700C': 0, 'ACS-MSN3800': 4, 'Mellanox-SN3800-D112C8': 4, 'ACS-MSN4700': 0}
-port_position_tuple_list = [(0, 0, 31, 32, 1), (0, 0, 15, 16, 1), (0, 48, 55, 56, 1), (0, 18, 21, 22, 1), (0, 0, 63, 64, 1)]
+hwsku_dict = {'ACS-MSN2700': 0, 'Mellanox-SN2700': 0, 'Mellanox-SN2700-D48C8': 0, "LS-SN2700":0, 'ACS-MSN2740': 0, 'ACS-MSN2100': 1, 'ACS-MSN2410': 2, 'ACS-MSN2010': 3, 'ACS-MSN3700': 0, 'ACS-MSN3700C': 0, 'ACS-MSN3800': 4, 'Mellanox-SN3800-D112C8': 4, 'ACS-MSN4700': 0, 'ACS-MSN3420': 5}
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py:44]: @shlomibitton why do we need this `"LS-SN2700":0` ?

**Pull request merged**